pyignite/stream/binary_stream.py

```diff
@@ -0,0 +1,114 @@
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import ctypes
+from io import BytesIO
+
+import pyignite.utils as ignite_utils
+
+READ_FORWARD = 0
+READ_BACKWARD = 1
+
+
+class BinaryStream:
+    def __init__(self, *args):
+        """
+        Initialize binary stream around buffers.
+
+        :param buf: Buffer, optional parameter. If not passed, creates empty BytesIO.
+        :param conn: Connection instance, required.
+        """
+        from pyignite.connection import Connection
+
+        buf = None
+        if len(args) == 1:
+            self.conn = args[0]
+        elif len(args) == 2:
+            buf = args[0]
+            self.conn = args[1]
+
+        if not isinstance(self.conn, Connection):
+            raise TypeError(f"invalid parameter: expected instance of {Connection}")
+
+        if buf and not isinstance(buf, (bytearray, bytes, memoryview)):
+            raise TypeError(f"invalid parameter: expected bytes-like object")
+
+        self.stream = BytesIO(buf) if buf else BytesIO()
+
+    @property
+    def compact_footer(self) -> bool:
+        return self.conn.client.compact_footer
+
+    @compact_footer.setter
+    def compact_footer(self, value: bool):
+        self.conn.client.compact_footer = value
+
```

Comment:
It looks like we only need a `conn` here for binary configuration parameters. Maybe it makes sense to make a separate class for that? Like, `BinaryContext`, `BinaryMetaProvider` or something like that?

Reply:
Original author use this property as auto-discovery of compact-footer for server.
I suppose that create separate class for 1 param seems like overengineering here.